synapse/handlers/federation.py

```diff
@@ -425,6 +472,11 @@ async def try_backfill(domains: List[str]) -> bool:
 
             return False
 
+        processing_end_time = self.clock.time_msec()
+        backfill_processing_before_timer.observe(
+            (processing_start_time - processing_end_time) / 1000
+        )
```

Comment:
Same problem as https://github.com/matrix-org/synapse/pull/13533#discussion_r949577244

I don't think this is working how we expect. All of the buckets just have the same values which means it's saying every request is taking more than 80s.

https://prometheus.matrix.org/graph?g0.expr=synapse_federation_backfill_processing_before_time_seconds_bucket&g0.tab=1&g0.stacked=0&g0.show_exemplars=0&g0.range_input=1h&g0.end_input=2022-08-18%2020%3A36%3A02&g0.moment_input=2022-08-18%2020%3A36%3A02
```
synapse_federation_backfill_processing_before_time_seconds_bucket{environment="live", identifier="matrix.org", index="2", instance="matrix.org", job="synapse_client_reader", le="+Inf", service="synapse"}	7759
synapse_federation_backfill_processing_before_time_seconds_bucket{environment="live", identifier="matrix.org", index="2", instance="matrix.org", job="synapse_client_reader", le="1.0", service="synapse"}	7759
synapse_federation_backfill_processing_before_time_seconds_bucket{environment="live", identifier="matrix.org", index="2", instance="matrix.org", job="synapse_client_reader", le="10.0", service="synapse"}	7759
synapse_federation_backfill_processing_before_time_seconds_bucket{environment="live", identifier="matrix.org", index="2", instance="matrix.org", job="synapse_client_reader", le="20.0", service="synapse"}	7759
synapse_federation_backfill_processing_before_time_seconds_bucket{environment="live", identifier="matrix.org", index="2", instance="matrix.org", job="synapse_client_reader", le="30.0", service="synapse"}	7759
synapse_federation_backfill_processing_before_time_seconds_bucket{environment="live", identifier="matrix.org", index="2", instance="matrix.org", job="synapse_client_reader", le="40.0", service="synapse"}	7759
synapse_federation_backfill_processing_before_time_seconds_bucket{environment="live", identifier="matrix.org", index="2", instance="matrix.org", job="synapse_client_reader", le="5.0", service="synapse"}	7759
synapse_federation_backfill_processing_before_time_seconds_bucket{environment="live", identifier="matrix.org", index="2", instance="matrix.org", job="synapse_client_reader", le="60.0", service="synapse"}	7759
synapse_federation_backfill_processing_before_time_seconds_bucket{environment="live", identifier="matrix.org", index="2", instance="matrix.org", job="synapse_client_reader", le="80.0", service="synapse"}	7759
```

https://grafana.matrix.org/d/dYoRgTgVz/messages-timing?orgId=1&from=1660812148100&to=1660855348100&viewPanel=212

![](https://user-images.githubusercontent.com/558581/185490897-41af4f7f-dbd5-4275-bdd1-a5cd0f16466b.png)


The after timing one looks normal:

https://grafana.matrix.org/d/dYoRgTgVz/messages-timing?orgId=1&from=1660812186785&to=1660855386785&viewPanel=213
![](https://user-images.githubusercontent.com/558581/185490981-ab3a5020-5d22-428a-b505-05ec599f024e.png)

Reply:
> Did we get the `ms` to `s` conversion wrong?

I don't think so. It gives values like `2.196`

Reply:
Spotted 🕵️‍♀️ I accidentally did `start - end` 🤦‍♂️

Corrected:

```suggestion
        processing_end_time = self.clock.time_msec()
        backfill_processing_before_timer.observe(
            (processing_end_time - processing_start_time) / 1000
        )
```

Reply:
Fixed in https://github.com/matrix-org/synapse/pull/13584